zipline/pipeline/factors/basic.py

```diff
@@ -489,6 +491,38 @@ def compute(self, today, assets, out, returns, annualization_factor):
         out[:] = nanstd(returns, axis=0) * (annualization_factor ** .5)
 
 
+class PeerCount(SingleInputMixin, CustomFactor):
+    """
+    Peer Count of distinct categories in a given classifier.  This factor
+    is returned by the classifier instance method peer_count()
+
+    **Default Inputs:** None
+
+    **Default Window Length:** 1
+    """
+    window_length = 1
+    missing_value = -1
+    dtype = int64_dtype
+
+    def _validate(self):
+        super(PeerCount, self)._validate()
+        if self.window_length != 1:
+            raise ValueError(
+                "'PeerCount' expected a window length of 1, but was given"
+                "{window_length}.".format(window_length=self.window_length)
+            )
+
+    def compute(self, today, assets, out, arrays):
```

Comment:
What do you think of calling `arrays` something more like `classifier_values` or `labels`?

Reply:
Definitely, changing it to classifier_values